src/ontoweaver/__init__.py

```diff
@@ -66,41 +80,58 @@ def extract_reconciliate_write(biocypher_config_path, schema_path, data_mappings
         schema_config_path = schema_path
     )
 
-    bc.write_nodes(fnodes)
-    bc.write_edges(fedges)
+    if fnodes:
+        bc.write_nodes(fnodes)
+    if fedges:
+        bc.write_edges(fedges)
     import_file = bc.write_import_call()
 
     return import_file
 
 
-def extract(data_mappings: dict, parallel_mapping = 0, affix="none", separator=":") -> Tuple[list[Tuple], list[Tuple]]:
+def extract(filename_to_mapping = None, dataframe_to_mapping = None, parallel_mapping = 0, affix="none", affix_separator=":") -> Tuple[list[Tuple], list[Tuple]]:
     """
     Extracts nodes and edges from tabular data files based on provided mappings.
 
     Args:
-        data_mappings (dict): a dictionary mapping data file path to the OntoWeaver mapping yaml file to extract them
+        filename_to_mapping (dict): a dictionary mapping data file path to the OntoWeaver mapping yaml file to extract them.
+        dataframe_to_mapping: a dictionary mapping loaded Pandas data frame to the loaded yaml mapping object.
         parallel_mapping (int): Number of workers to use in parallel mapping. Defaults to 0 for sequential processing.
-        separator (str, optional): The separator to use for splitting ID and type. Defaults to None.
         affix (str, optional): The affix to use for type inclusion. Defaults to "none".
         affix_separator: The character(s) separating the label from its type affix. Defaults to ":".
 
     Returns:
         tuple: Two lists of tuples containing nodes and edges.
     """
 
-    assert(type(data_mappings) == dict) # data_file => mapping_file
-
     nodes = []
     edges = []
+    adapter = None
+
+    if filename_to_mapping:
+
+        assert(type(filename_to_mapping) == dict) # data_file => mapping_file
+
+        for data_file, mapping_file in filename_to_mapping.items():
+            table = pd.read_csv(data_file)
+
+            with open(mapping_file) as fd:
+                mapping = yaml.full_load(fd)
+
+            adapter = tabular.extract_table(table, mapping, parallel_mapping=parallel_mapping, affix=affix,
+                                            separator=affix_separator)
+
+
+    if dataframe_to_mapping:
 
-    for data_file, mapping_file in data_mappings.items():
-        table = pd.read_csv(data_file, sep = None)
+        assert(type(dataframe_to_mapping) == dict) # data_frame => yaml_object
 
-        with open(mapping_file) as fd:
-            mapping = yaml.full_load(fd)
+        for data_frame, yaml_object in dataframe_to_mapping.items():
 
-        adapter = tabular.extract_all(table, mapping, parallel_mapping=parallel_mapping, affix=affix, separator=separator)
+            adapter = tabular.extract_table(data_frame, yaml_object, parallel_mapping=parallel_mapping, affix=affix,
+                                            separator=affix_separator)
```

Comment:
This is going to overwrite the adapter object, and not accumulate the nodes. This is my fault with this bad idea of putting the node accumulation in a later section. Sorry.